src/compiler/core.ts

```diff
@@ -1222,6 +1242,13 @@ namespace ts {
         return Array.isArray ? Array.isArray(value) : value instanceof Array;
     }
 
+    /**
+     * Tests whether a value is string
+     */
+    export function isString(text: any): text is string {
+        return typeof text === "string";

```

Comment:
This is likely a minor slow-down, for no benefit to readability.

Functions that are called with polymorphic arguments are quickly de-optimised. This one is specifically intended to be called with polymorphic arguments, so it's very likely to be de-optimised and become slower than conventional idiomatic `typeof === 'string'`.

Worth checking with a timed run.